task_health_service.go

```diff
@@ -0,0 +1,65 @@
+package influxdb
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+// FindTasksService implements the FindTasks method of a TaskService for the TaskHealthService to use
+type FindTasksService interface {
+	FindTasks(ctx context.Context, filter TaskFilter) ([]*Task, int, error)
+}
+
+// TaskHealthService is responsible for monitoring the latency of active tasks in the kv store
+type TaskHealthService struct {
```

Comment:
I would recommend we limit the responsibility of this service to almost exactly what you state in the comment. We might consider being explicit about the name:

> `TaskLatencyMetricsService` is responsible for determining the run latency of active tasks.

I would expose two methods for this service:

```
UpdateMetrics(ctx context.Context) error
```

Which is responsible for calling find tasks and updating the metrics

and a function like to `storage.Engine#PrometheusCollectors`:

https://github.com/influxdata/influxdb/blob/8f6701d4b1a12c7a5dd2f0adc92ea5ab4967e5f8/storage/engine.go#L220-L228

To return the global metrics.

I would add a second API, `ExtendedPrometheusCollectors` to return the metrics with task_id labels, which will have much higher cardinality. We can add these collectors to a separate HTTP end point, like `/metrics/tasks/usage`, so they can be queried manually but will not be gathered by telegraf by default.

By limiting the concern of this type to gathering metrics of latencies, it will allow us to determine how best to incorporate the type in OSS and Cloud.